### DISCUSSION TITLE
Performance Comparison Mac Build Host Pro vs Air

### DISCUSSION THREAD

**Discussion by maslewandowski** [dotnet/maui#24815] [General]
Hi there,
unfortunately i don't have the opportunity to compare the performance of a macbook air against a macbook pro.

I am currently using a Macbook Air with a M2 cpu.

The build time with cabel bound network connection is.... long :)
I developed an application for my company and it takes about 2-4 mins to build it in release build config to distribute the app.

Is that a normal timespan?

Does anyone have any experience with the difference in build time with a Macbook Air M2 compared to a Macbook Pro M2 Max / M3?

Probably thermal throttling is a big thing here.

thanks and greetings

**Comment by MartyIX** [comment]
Would this https://github.com/dotnet/maui/discussions/24299#discussioncomment-10398098 help you debug it?

**Comment by Toine-db** [comment]
@maslewandowski  Have you already discovered if it can work well on your M2?
I am also looking at whether I should buy an old airbook or a new Mini 4 for this

**Reply by maslewandowski** [reply]
i'm still using m2 airbook and if i build my application (200 mb) the build host needs 14+ minutes to build for release (bundle signing etc.)
but while developing the time is much shorter. like 1-2 mins max

i do not use antivirus software and the network connection is via ethernet.

so yeah i would love to see if there is a build time difference between macbook air m2 and the max models

<img width="268" alt="image" src="https://github.com/user-attachments/assets/83f55afe-1e7b-46a1-abb2-800dc30cadcc">

**Reply by Toine-db** [reply]
I currently still are using a Intel Mac Mini, so I can't give you that measurement.
But I will try to measure a Debug build on the Intel device, just to check.
